File: tmdb-core/src/main/java/dev/reuss/tmdb/domain/tv/series/model/TvSeriesVideos.java
package dev.reuss.tmdb.domain.tv.series.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import dev.reuss.tmdb.common.TmdbModel;
import dev.reuss.tmdb.common.video.VideosResponse;
import dev.reuss.tmdb.util.TmdbCollections;

import java.util.List;

/**
 * Videos for a TMDB TV series.
 *
 * @param id      TMDB TV series id
 * @param results TV series videos
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record TvSeriesVideos(
        int id,
        List<TvSeriesVideo> results
) implements VideosResponse<TvSeriesVideo>, TmdbModel {

    public TvSeriesVideos {
        results = TmdbCollections.list(results);
    }
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/domain/tv/series/model/TvSeriesVideos.kt
package dev.reuss.tmdb.domain.tv.series.model

import com.fasterxml.jackson.annotation.JsonProperty
import dev.reuss.tmdb.common.TmdbModel
import dev.reuss.tmdb.common.video.VideosResponse

/**
 * Videos for a TMDB TV series.
 *
 * @property id TMDB TV series id
 * @property results TV series videos
 */
@JvmRecord
data class TvSeriesVideos(
    @all:JsonProperty("id")
    val id: Int,
    @all:JsonProperty("results")
    override val results: List<TvSeriesVideo> = emptyList(),
) : VideosResponse<TvSeriesVideo>,
    TmdbModel
